src/main/java/com/flightstats/hub/dao/aws/S3BatchManager.java

```diff
@@ -65,8 +65,9 @@ private void setupBatch() {
                     appProperties.getAppEnv());
 
             if (channel.isSingle()) {
-                if (!webhookLeaderState.getServers(channel.getName()).isEmpty()) {
-                    log.debug("turning off batch webhook {}", channel.getDisplayName());
+                WebhookLeaderState.RunningState state = webhookLeaderState.getState(channel.getName());
+                if (!state.isStopped()) {
+                    log.debug("turning off batch webhook for {}", channel.getDisplayName());
```

Comment:
why is there a s3batch.delete if the channel.isSingle?

Reply:
this is unrelated to your changes, I'm just not sure what is going on here

Reply:
I believe it's checking to see if the channel changed from batch (or both?) to single storage mode (single channel, but the batch webhook is still running) and stops the webhook if it's in that state.

Reply:
that may not be entirely accurate. I knew what this did at one point, I think.